testsuite/pytests/test_stdp_synapse.py

```diff
@@ -209,42 +224,49 @@ def Kpost_at_time(t, spikes, inclusive=True):
             Kpost *= exp(-(t - t_curr) / self.tau_post)
             return Kpost
 
+        eps = 1e-6
         t = 0.
+        idx_next_pre_spike = 0
+        idx_next_post_spike = 0
+        t_last_pre_spike = -1
+        t_last_post_spike = -1
         Kpre = 0.
         weight = initial_weight
 
-        t_log = []
-        w_log = []
-        Kpre_log = []
-
-        # logging
-        t_log.append(t)
-        w_log.append(weight)
-        Kpre_log.append(Kpre)
+        t_log = list()
+        w_log = dict()
```

Comment:
This can just stay a list, right? The timepoint corresponding to ``w[i]`` is ``t[i]`` for all *i*.

Reply:
Yup, you are completely right!